http/etag.ts

```diff
@@ -0,0 +1,196 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+/** Provides functions for dealing with and matching ETags, including
+ * {@linkcode calculate} to calculate an etag for a given entity,
+ * {@linkcode ifMatch} for validating if an ETag matches against a `If-Match`
+ * header and {@linkcode ifNoneMatch} for validating an Etag against an
+ * `If-None-Match` header.
+ *
+ * See further information on the `ETag` header on
+ * [MDN](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag).
+ *
+ * @module
+ */
+
+import { type DigestAlgorithm } from "../crypto/crypto.ts";
+import { toHashString } from "../crypto/to_hash_string.ts";
+import { createHash } from "../crypto/_util.ts";
```

Comment:
This introduces the dependency to the entire `std/crypto`. I feel that is too excessive for this module as hash algorithm doesn't matter very much for this module in my view. How about just using web crypto API for creating hash?

Reply:
Fair point. I ported the stuff from file server, so there was consistency, but I agree it doesn't make a huge amount of sense.